internal/kubernetes/reflectionController.go

```diff
@@ -112,6 +112,18 @@ func (p *KubernetesProvider) controlLoop() {
 
 func (p *KubernetesProvider) cleanupNamespace(ns string) error {
 
+	pods, err := p.foreignClient.Client().CoreV1().Pods(ns).List(metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	for _, po := range pods.Items {
+		err = p.foreignClient.Client().CoreV1().Pods(ns).Delete(po.Name, &metav1.DeleteOptions{})
```

Comment:
Before deleting all the pods, I think we should set the virtual node as `notReady`, otherwise, if we delete the foreign pod, the vk, will notify this event and will delete also the local copy, leading to the `replicaController` trigger that re-creates the pod, scheduling it again to the virtual node.

Reply:
This comment is related to this line, even though the node status setting should be performed while detecting the advertisement deletion [here](https://github.com/LiqoTech/liqo/blob/4b29ac14aa1f67de0ded0a248a2f5983218eac3a/internal/kubernetes/virtualNodeupdater.go#L59).

Reply:
thanks!